README.md

```
@@ -8,155 +8,58 @@
-MooseStack offers a unified, type‑safe, code‑first developer experience layer for [ClickHouse](https://clickhouse.com/) (realtime analytical database), [Kafka](https://kafka.apache.org/)/[Redpanda](https://redpanda.com/) (realtime streaming), and [Temporal](https://temporal.io/) (workflow orchestration). So you can integrate real time analytical infrastructure into your application stack in native Typescript or Python.
+Generalist coding agents like Claude Code, Cursor, and Copilot work well for most coding tasks. But ClickHouse and realtime analytical infrastructure are specialized terrain. MooseStack sits between your agent and your data stack, and enables your agent to handle the stickiest parts of building out applications on real time analytical infra. Think:
+- Optimizing data models & queries for OLAP
+- Managing schema changes and migrations
+- Building out realtime streaming ETL/ELT data pipelines
+- Integrating analytics into your frontend application layer
 
-Because your entire analytical stack is captured in code, AI agents can read, write, and refactor it just like any other part of your application. Combined with fast local feedback loops via `moose dev` and complementary [ClickHouse best-practice agent skills](https://github.com/514-labs/agent-skills), agents can often handle the bulk of your development work — schema design, materialized views, migrations, and query APIs — while you review and steer.
+![MooseStack agent harness diagram](./harness.png)
 
-## Why MooseStack?
+## How it works
```

markdownlint-cli2 (0.21.0)

[warning] Headings should be surrounded by blank lines
Expected: 1; Actual: 0; Below

(MD022, blanks-around-headings)